sycl/doc/extensions/experimental/sycl_ext_intel_gpu_cache_config.asciidoc

```diff
@@ -0,0 +1,177 @@
+= sycl_ext_intel_gpu_cache_config
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Notice
+
+[%hardbreaks]
+Copyright (c) 2023-2023 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+https://github.com/intel/llvm/issues
+
+== Contributors
+
+Greg Lueck, Intel +
+John Pennycook, Intel +
+Artur Gainullin, Intel
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 6 and
+the following extensions:
+
+- link:../experimental/sycl_ext_oneapi_properties.asciidoc[sycl_ext_oneapi_properties]
+- link:sycl_ext_oneapi_kernel_properties.asciidoc[sycl_ext_oneapi_kernel_properties]
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback. Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state. The specification itself may also change in
+incompatible ways before it is finalized. *Shipping software products should not
+rely on APIs defined in this specification.*
+
+== Overview
+
+There are devices where the same hardware resources are used for shared local
+memory (SLM) and data cache. Developers may want more data cache or SLM based
+on their application. This extension adds runtime kernel property `gpu_cache_config`
```

Comment:
SLM is also for some "data", right? How people decide which they need more SLM or Data?

Reply:
I used the same wording "SLM" and "data" as in the L0 spec: https://spec.oneapi.io/level-zero/latest/core/PROG.html?highlight=sets%20preferred%20cache%20configuration#device-cache-settings

If kernel allocates much local memory then user may decide to go with large_slm. If small amount of local memory is allocated in the kernel then user may use large_data having more cache available for data which is not in local memory.

Reply:
I suggest writing this spec more clear than in L0 spec ) And maybe also name them more clear

Reply:
@againull : maybe some of the explanation from https://www.intel.com/content/www/us/en/docs/oneapi/optimization-guide-gpu/2023-0/shared-local-memory.html could be used here

Reply:
Updated documentation with more details.